recipes/fmt/all/conanfile.py

```diff
@@ -30,29 +35,27 @@ class FmtConan(ConanFile):
         "with_os_api": True,
     }
 
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
     @property
     def _has_with_os_api_option(self):
-        return tools.Version(self.version) >= "7.0.0"
+        return Version(str(self.version)) >= "7.0.0"
+
+    def generate(self):
+        if not self.options.header_only:
+            tc = CMakeToolchain(self)
+            tc.variables["FMT_DOC"] = False
+            tc.variables["FMT_TEST"] = False
+            tc.variables["FMT_INSTALL"] = True
+            tc.variables["FMT_LIB_DIR"] = "lib"
+            if self._has_with_os_api_option:
+                tc.variables["FMT_OS"] = self.options.with_os_api
+            tc.generate()  
+
+    def layout(self):
+        cmake_layout(self)
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        for _patch in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, _patch["patch_file"], src=".", dst=self.export_sources_folder)
```

Comment:
You should be able to simplify this with `apply_conandata_patches(self)`

Reply:
In fact, I'll remove this method, I think that they are patching twice in the original recipe? exports_sources and build ?